starcounter-include.html

```diff
@@ -284,6 +285,17 @@
                     this.partialId = null;
                 }
 
+                // Find parent composition to clone (if have one)
+                const parentCompositionTemplate = Array.from(this.children)
+                    .find((element)=>{
+                        return element.matches
+                            && element.matches('template[is="declarative-shadow-dom"][presentation="parent"]')
```

Comment:
I like the code now!

My only doubt is that we agreed that the parent composition `declarative-shadow-dom` can be determined by heuristics of being a child of `starcounter-include`.

In your implementation though, it is determined by explicit attribute. Do you think this is unavoidable for future compatibility with flattened views?

I think it is avoidable. The "flattener" could insert `presentation="default"` attribute automatically for default presentations when it is flattening them in DOM. This way we would know which `declarative-shadow-doms` are parent and which not.

Reply:
Right now, we can. For flattened we will need a way to differentiate them.

Reply:
>  think it is avoidable. The "flattener" could insert presentation="default" attribute automatically for default presentations when it is flattening them in DOM.

Reply:
Also to do that, we would have to parse the HTML, and may not cover the cases when `<tempalte is="declaratice-shadow-dom">` is added dynamically by some script.

Reply:
If `presentation` attribute is not provided, we can set both `presentation="default"` and `presentation="parent"` implicitly by `s-i`, based on what `s-i` thinks it is.

So your example from https://github.com/Starcounter/starcounter-include/pull/103#discussion_r198749151 becomes:

```html
<starcounter-include>
  #shadow-root //let's find out where it came from?
  <hundrets>
  <of>
  <elements>
  <template is="declarative-shadow-dom" presentation="default">...</template>
  <hundrets>
  <of>
  <elements>
  <template is="declarative-shadow-dom" presentation="default">...</template>
  <hundrets>
  <of>
  <elements>
  <template is="declarative-shadow-dom" presentation="default">...</template>
  <hundrets>
  <of>
  <elements>
  <template is="declarative-shadow-dom" presentation="parent">...</template>
</starcounter-include>
```